lib/puppet/provider/service/smf.rb

```diff
@@ -121,33 +148,26 @@ def restart
     self.wait('online')
   end
 
-  # Determine the current and next states of a service.
-  def service_states
-    svcs("-H", "-o", "state,nstate", @resource[:name]).chomp.split
-  end
-
   def status
-    if @resource[:status]
-      super
-      return
-    end
+    # TODO (PUP-8957): This originally returned nil after calling super, which is inconsistent
+    # with how things are done.
+    return super if @resource[:status]
 
     begin
-      # get the current state and the next state, and if the next
-      # state is set (i.e. not "-") use it for state comparison
-      states = service_states
-      state = states[1] == "-" ? states[0] : states[1]
+      # Get the current state and the next state. If there is a next state,
+      # use that for the state comparison.
+      states = self.service_states
+      state = states[:next] || states[:current]
     rescue Puppet::ExecutionFailure
+      # TODO (PUP-8957): Should this be set back to INFO ?
       debug "Could not get status on service #{self.name} #{$!}"
-      return :absent
+      return :stopped
```

Comment:
So if we can't get a state, we indicate its not running? Is it possible for us not to be able to query it and get this same failure condition (indicating absent as previously implemented?)

Reply:
One of the key parts of this work is to remove the `:absent` thing b/c it clutters people's logs (it's why the original PE ticket was filed -- https://tickets.puppetlabs.com/browse/PE-24785. Idea being is if you have a manifest `ensure => stopped` for a nonexistent service, then the `is` value is `:absent` while the `should` value is `:stopped`, so Puppet will output noise saying your param ensure's changed from `:absent` to `:stopped`, which is wrong (+ ensure has only two values here -- https://github.com/puppetlabs/puppet/blob/5.5.x/lib/puppet/type/service.rb#L98-L104 so it is wrong of this code to return other states as well, since `status` is `ensure`'s retrieve method. But I don't want to mess with the underlying interface too much since that wasn't an issue in the past).

Reply:
Thats what I was thinking -- changing state to absent made no sense in this context.